packages/dds/tree/src/simple-tree/schemaTypes.ts

```diff
@@ -251,22 +247,36 @@ export class FieldSchema<
 		 */
 		public readonly allowedTypes: Types,
 	) {
-		this.lazyTypes = new Lazy(() => FieldSchema.normalizeAllowedTypes(this.allowedTypes));
+		this.lazyTypes = new Lazy(() => normalizeAllowedTypes(this.allowedTypes));
 	}
+}
 
-	/**
-	 * Normalizes a {@link ImplicitAllowedTypes} to a {@link AllowedTypes}.
-	 */
-	public static normalizeAllowedTypes(types: ImplicitAllowedTypes): AllowedTypes {
-		return isReadonlyArray(types) ? types : [types];
+/**
+ * Normalizes a {@link ImplicitFieldSchema} to a {@link FieldSchema}.
+ */
+export function normalizeFieldSchema(schema: ImplicitFieldSchema): FieldSchema {
+	return schema instanceof FieldSchema ? schema : new FieldSchema(FieldKind.Required, schema);
+}
+/**
+ * Normalizes a {@link ImplicitAllowedTypes} to a {@link AllowedTypes}.
+ */
+export function normalizeAllowedTypes(types: ImplicitAllowedTypes): ReadonlySet<TreeNodeSchema> {
```

Comment:
This comment and the signature don't match.

I'm fine with this utility doing either as long as the docs match.

Reply:
Actually, if this does keep returning the set, you might want to put a warning in the comment noting that this evaluates lazy schema references and thus can not be used until all schema are finished being declared to avoid issues with recursive schema. Might also be good to check what symptoms using this when invalid causes and throw a better error than however it would otherwise break (I think it would put an undefined in the set, which woul dbe nasty to debug)

Reply:
```suggestion
 * Normalizes a {@link ImplicitAllowedTypes} to a set of {@link TreeNodeSchema}s.
 */
export function normalizeAllowedTypes(types: ImplicitAllowedTypes): ReadonlySet<TreeNodeSchema> {
```

Reply:
I added a couple of simple unit tests, including a case where an implicit field schema comes in `undefined`. I updated the lazy evaluation flow to explicitly check for this and throw an error.